test/tests/unit/probe/router.test.ts

```diff
@@ -447,37 +447,73 @@ describe('probe router', () => {
 			expect(probes[0]!.location.country).to.equal('GB');
 		});
 
-		it('should return result sorted by priority of magic fields', async () => {
+		it('should return result sorted by priority of magic fields in case of partial match', async () => {
 			const sockets: Array<DeepPartial<Socket>> = [
-				await buildSocket('socket-3', { country: 'PL', normalizedCity: 'warsaw', normalizedNetwork: 'ultra development networks' }),
+				await buildSocket('socket-3', { country: 'CZ', normalizedCity: 'praga', normalizedNetwork: 'ultra development networks' }),
 				await buildSocket('socket-2', { country: 'RS', normalizedCity: 'belgrade', normalizedNetwork: 'belgrade networks' }),
 				await buildSocket('socket-1', { country: 'DE', normalizedCity: 'berlin', normalizedNetwork: 'berlin networks' }),
 			];
 			fetchSocketsMock.resolves(sockets as never);
 
 			const probes = await router.findMatchingProbes([
-				{ magic: 'de' },
+				{ magic: 'd' },
 			], 100);
 
 			expect(fetchSocketsMock.calledOnce).to.be.true;
 			expect(probes.length).to.equal(3);
 			expect(probes[0]!.location.country).to.equal('DE');
 			expect(probes[1]!.location.country).to.equal('RS');
+			expect(probes[2]!.location.country).to.equal('CZ');
+		});
+
+		it('should ignore matches on the lower levels after exact match', async () => {
+			const sockets: Array<DeepPartial<Socket>> = [
+				await buildSocket('socket-3', { country: 'VN', normalizedCity: 'hanoi', normalizedNetwork: 'ultra networks' }),
+				await buildSocket('socket-2', { country: 'RU', normalizedCity: 'vnukovo', normalizedNetwork: 'super networks' }),
+				await buildSocket('socket-1', { country: 'HU', normalizedCity: 'budapest', normalizedNetwork: '23VNet Kft' }),
+			];
+			fetchSocketsMock.resolves(sockets as never);
+
+			const probes = await router.findMatchingProbes([
+				{ magic: 'vn' },
+			], 100);
+
+			expect(fetchSocketsMock.calledOnce).to.be.true;
+			expect(probes.length).to.equal(1);
+			expect(probes[0]!.location.country).to.equal('VN');
+		});
+
+		it('should accept partial matches on the higher or same level as exact match', async () => {
+			const sockets: Array<DeepPartial<Socket>> = [
+				await buildSocket('socket-1', { country: 'PL', normalizedCity: 'warsaw', normalizedNetwork: 'wars' }),
+				await buildSocket('socket-2', { country: 'PL', normalizedCity: 'warsaw', normalizedNetwork: 'super networks' }),
+				await buildSocket('socket-3', { country: 'PL', normalizedCity: 'poznan', normalizedNetwork: 'wars networks' }),
+			];
+			fetchSocketsMock.resolves(sockets as never);
+
+			const probes = await router.findMatchingProbes([
+				{ magic: 'wars' },
+			], 100);
+
+			expect(fetchSocketsMock.calledOnce).to.be.true;
+			expect(probes.length).to.equal(3);
+			expect(probes[0]!.location.country).to.equal('PL');
+			expect(probes[1]!.location.country).to.equal('PL');
 			expect(probes[2]!.location.country).to.equal('PL');
 		});
```

Comment:
Sorry highlighted the wrong test before:

This matches what I described in my last comment, but now I'm thinking it would make more sense even here to include only the exact match (https://github.com/jsdelivr/globalping/issues/247#issuecomment-1339815722). The impact isn't big either way, though, as this is most likely going to happen only with networks.

Reply:
Note that a more interesting scenario with this problem may be:

```js
const sockets: Array<DeepPartial<Socket>> = [
	await buildSocket('socket-1', { country: 'PL', normalizedCity: 'warsaw', normalizedNetwork: 'super networks' }),
	await buildSocket('socket-2', { country: 'PL', normalizedCity: 'warsaw', normalizedNetwork: 'super networks' }),
	await buildSocket('socket-3', { country: 'PL', normalizedCity: 'poznan', normalizedNetwork: 'wars' }),
];
```

With the current version of this PR and `limit: 1`, it would always return one of the Warsaw probes because cities have higher priority than networks. If we change it to "exact match always wins", it would return Poznan in this case.

Reply:
More realistic examples that still don't work as expected are cities "Oi" vs "Hanoi" and "New York City" vs "York".

Reply:
We can either do "if there's an exact match, ignore partial matches" or "give the exact matches highest priority over index fields priorities".

Reply:
Let's go with first, regardless of sorting. Sorting will be used only if there's no exact match.

Reply:
done